PR Review for Component Library (v0.0 alpha release)

codingwolf-at opened the pull request:
Name: Atul Tameshwari
Live link: https://crescent-ui-alpha.netlify.app/

Note: This is an alpha release i.e., not all components are added and added components still need some polishing. I will keep adding new components and polishing existing components throughout the camp.

codingwolf-at committed:
initial commit

codingwolf-at committed:
added: card pngs

codingwolf-at committed:
added: home key png

codingwolf-at committed:
added: badge and buttons section

codingwolf-at committed:
deleted: redundant file

codingwolf-at committed:
deleted: redundant file

codingwolf-at committed:
added: card section

codingwolf-at committed:
feat: implemented fixed side nav

codingwolf-at committed:
style: side nav

codingwolf-at committed:
fix: typo

codingwolf-at committed:
added: card code

codingwolf-at committed:
fix: updated button and badge code snippets

codingwolf-at committed:
fix: updated alert code snippet

codingwolf-at committed:
fix: updated avatar code snippet

codingwolf-at committed:
style: scroll behavior

codingwolf-at committed:
added: navbar section

codingwolf-at committed:
added: spacing utilities section

codingwolf-at committed:
added: typography section

codingwolf-at committed:
added: images

codingwolf-at committed:
added: images section

codingwolf-at committed:
updated: side nav links

codingwolf-at committed:
updated: second card bg color

codingwolf-at committed:
fix: img border radius

MdBilal420 commented:
Hi,
1. Cart component name and price should have different fonts.(sir told this in code review)
2. Navbar component, on site-logo cursor is not pointer and demo navbar link are not href="#" they are redirecting.
3. Alert component is very cool.

codingwolf-at commented:
> Hi,
> 
> 1. Cart component name and price should have different fonts.(sir told this in code review)
> 2. Navbar component, on site-logo cursor is not pointer and demo navbar link are not href="#" they are redirecting.
> 3. Alert component is very cool.

Thank you for your feedback!
I have changed the cursor property on the brand logo and also changed the price font. 
But about the demo nav link, they are already having href="#" in all the four demo navbars, you can confirm that from the code snippet below the component.

codingwolf-at committed:
removed redundant code

codingwolf-at committed:
added: highlight js cdn

codingwolf-at committed:
fix: replaced all iframes with pre and code tag

codingwolf-at committed:
style: pre tag

codingwolf-at committed:
style: changed header bg color

SurajGupta07 commented:
The code looks neat and overall great component library.

Xdev200 commented on css/docs.css:
```diff
@@ -0,0 +1,155 @@
+code {
+  border-radius: 0.5rem;
+}
+
```
pre and code tags are not supported in HTML5

codingwolf-at replied:
I checked and it turns out that they are supported.
[HERE](https://www.w3.org/TR/2011/WD-html5-author-20110809/the-code-element.html) is the W3C link regarding this, please check it out.

Xdev200 commented on css/index.css:
```diff
@@ -0,0 +1,90 @@
+/* Font Import */
+@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;600&display=swap');
+
```
Not A good Practice to include import in CSS file because code below the import code will execute after all the imports get load.

codingwolf-at replied:
Done.
Included google font's <link> in HTML file

Xdev200 commented on css/index.css:
```diff
@@ -0,0 +1,90 @@
+    background-position: center;
+    background-repeat: no-repeat;
+    background-size: cover;
+    height: 100vh;  
```
You can use short properties for background 
background: url(/img/bg-one_1280x800.jpg) no-repeat center cover;

codingwolf-at replied:
Yes, I changed it.
But the shorthand example you provided didn't work.

According to MDN and W3, there needs to be a slash separating the background-size from the background-position.
[Link regarding this](https://stackoverflow.com/a/17093346)

The correct syntax that worked is : 
`background: url(/img/bg-one_1280x800.jpg) center/cover no-repeat;`

codingwolf-at commented:
> The code looks neat and overall great component library.

Thank you for your feedback Suraj

codingwolf-at committed:
refactor: background and font

MinakshiMittal commented:
The landing page is awesome!

Few suggestions:

- For the close buttons instead of using 'x' you can use the font-awesome icon. `<i class="fas fa-lg fa-times"></i>`
 
- https://github.com/codingwolf-at/crescent-ui/blob/4ffa124e80f65a98778bc1a453ac60f3eafcd145/docs.html#L402-L404
When you click on any of the links it leads you to the top of the page. You can add `href="javascript:void(0);` it will prevent that.

codingwolf-at committed:
refactor: replaced href="#" by javascript:void(0)

codingwolf-at commented:
> The landing page is awesome!
> 
> Few suggestions:
> 
> * For the close buttons instead of using 'x' you can use the font-awesome icon. `<i class="fas fa-lg fa-times"></i>`
> * https://github.com/codingwolf-at/crescent-ui/blob/4ffa124e80f65a98778bc1a453ac60f3eafcd145/docs.html#L402-L404
>   
>   When you click on any of the links it leads you to the top of the page. You can add `href="javascript:void(0);` it will prevent that.

Hey, Minakshi thanks for your feedback.
I fixed all the `<a>` tags 👍
And regarding the close button, I don't wanna use one more library inside my own library so I think I'm gonna stick with the "X" as of now. But let me know if there is any "vanilla" way of doing it (without using any other library).

codingwolf-at committed:
added: mobile background

codingwolf-at committed:
removed: redundant code

codingwolf-at committed:
fix: main-page-div margin issue

codingwolf-at committed:
deleted file

codingwolf-at committed:
refactor: removed one unused class

codingwolf-at committed:
initial commit

codingwolf-at committed:
refactor: removed redundant class on right div

codingwolf-at committed:
style: added padding to right div

codingwolf-at committed:
done with mobile view

codingwolf-at committed:
style: top margin for right div

codingwolf-at committed:
fix: margin-top for right div

codingwolf-at committed:
refactor: removed redundant classes

codingwolf-at committed:
style: header and main element padding

codingwolf-at committed:
initial commit

codingwolf-at committed:
added: margin-left to logo

codingwolf-at committed:
style: header for mobile view

codingwolf-at committed:
style: header

codingwolf-at committed:
style: main context padding and pre tag padding

codingwolf-at committed:
style: overriden some crerscent classes for mobile

codingwolf-at committed:
style: overridden nav class

Amarjit-Singh-1 commented:
Great Work! Your Documentation as well as the Code is Awesome.
Few Suggestions:
1. Badges : The Badges on the Avatar are very small in size , you can increase its font-size . Also the one with badge value 0 looks a little oval in shape
2. Cards : You can align the text in the Cards to center which would look better . You can use the class="center" on h2 and h4.

codingwolf-at commented:
> Great Work! Your Documentation as well as the Code is Awesome.
> Few Suggestions:
> 
> 1. Badges : The Badges on the Avatar are very small in size , you can increase its font-size . Also the one with badge value 0 looks a little oval in shape
> 2. Cards : You can align the text in the Cards to center which would look better . You can use the class="center" on h2 and h4.

Thansk Amarjit for your feedback, 
1) yes, I will fix the badges.
2) I personally prefer left-aligned text inside my card, but thanks for suggesting centered text I'll think about it.

codingwolf-at committed:
docs: updated README

codingwolf-at committed:
docs: updated README

codingwolf-at committed:
docs: README updated

kushanksriraj commented on css/docs.css:
```diff
@@ -0,0 +1,159 @@
+
+.highlight {
+  background-color: var(--grey);
+  font-family: monospace;
```
Avoid changing fonts.

codingwolf-at replied:
Is this a bad practice? I searched about it and couldn't find anything. 
And if this is, then how else can I modify the font in a particular place if I need to?

kushanksriraj commented on css/docs.css:
```diff
@@ -0,0 +1,159 @@
+  font-family: monospace;
+}
+
+pre {
```
Don't style tags directly, use class for it.

kushanksriraj commented on css/docs.css:
```diff
@@ -0,0 +1,159 @@
+  position: fixed;
+  height: 100%;
+  overflow: auto;
+  top: 0%;
```
`top: 0;` is better as you don't need to specify the CSS unit.

codingwolf-at replied:
fixed

kushanksriraj commented on css/docs.css:
```diff
@@ -0,0 +1,159 @@
+:root {
+  /* color */
+  --header-bg: #1F2937;
+  --light-grey: #F1F3F5;
```
Your variable should specify its purpose: `--light-grey-color`.

codingwolf-at replied:
fixed

kushanksriraj commented on index.html:
```diff
@@ -0,0 +1,59 @@
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Montserrat:wght@400;600&display=swap" rel="stylesheet">
+    <!-- crescent ui components stylesheet -->
+    <link rel="stylesheet" href="https://crescent-components.netlify.app/css/crescent.css">
```
Unable to find this CSS file to review.

codingwolf-at committed:
fixed variable names and removed redundant code

codingwolf-at committed:
FIX: typo in image section
